### PR TITLE
Add Makefile

### LINKED ISSUE #1864: make Makefile - start development serve and application with a single command.
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behaviour:**

Currently to run the application (locally), I need to open 1 terminal for development server, the other for starting ember server. Also if I need to run the test of both frontend and backend, I need to check their syntax, and then type both commands seperately - burden for the developer.
Instead a one stop solution can be using Makefile instead.
To run both servers, we can use Procfile.dev (need to create one), and then use tools like (honcho)[https://github.com/nickstenning/honcho] or (foreman)[http://ddollar.github.io/foreman/] . All this will be started using single command `make start`.
Though we can use honcho directly, but there are many benifits of using Makefile - 

* Can be scaled up for use in production easily - just had to switch from Procfile.dev to Procfile
* Running tests easily
* Cleaning up the setup - completely
* Installing both frontend and backend requirements with a single command.
* No problem of learning new commands for basic but important tasks.

**For developers/new users**:
`make install`
`make start`
will be the only commands to start developing.

@gabru-md @ParthS007 @yashLadha @djmgit  your opinion on this?

**Would you like to work on the issue?**

**Yes**

### PR DESCRIPTION
**Pull request by rupav**
Fixes #1864 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added a Makefile, honcho to start the ember and flask server together. using `make start` command only. 
- `make install` installs both backend and frontend requirements, except other dependencies (like postgres/bower).

Screenshots of working (`make start` command to start both services):
screenshot from 2018-12-08 01-08-02
screenshot from 2018-12-08 01-09-10

Will add tests command too.

@ParthS007 @gabru-md @rajvaibhavdubey @yashLadha @djmgit please review.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+clean.frontend:
+	rm -rf $(NODE_MODULES)
+
+clean:
+	make clean.frontend
+
+install:
+	$(PIP) install --ignore-installed -r $(API)/requirements.txt
+	$(NPM) --prefix $(FRONTEND) install
+
```
Comment (rajvaibhavdubey) [inline Makefile]: Bower install also needs to be added

**Comment by ParthS007** [comment]
@rupav LGTM, Just add bower install and its good to merge. :+1:

**Comment by rupav** [comment]
done @ParthS007 @rajvaibhavdubey . Used https://github.com/bower/bower/issues/212#issuecomment-62998828 to install from frontend folder.

File: Makefile (hunk 1 above)
Comment (rupav) [reply Makefile]: DOne

**Comment by rupav** [comment]
@ParthS007 add `make test` command as well. - will run only frontend tests for now.

**Pull request merged**